### ISSUE TITLE
World Record Badges not correlated

### ISSUE THREAD

**Issue by ph4r** [SauceLLC/sauce4strava#62]
I did two Zwift races this weekend, and the segment results are at most Cat5 Badged, but the Analysis Page shows Cat4 on each of them.
[Sunday](https://www.strava.com/activities/6403169024)
[Saturday](https://www.strava.com/activities/6398521660)
I assume this is just matching to some Coggan Tables, but I'm not typically sure how the overall would be higher than the parts.

**Comment by mayfield** [comment]
Hi @ph4r probably it's just the time domain and spatial bounds of the segments vs the peak power table which is unbounded spatially.  I just goes through the power stream looking for the highest power averages of each time period.  For example, while it's possible that your peak 1 minute power happened to line up exactly with a 1 minute segment it is unlikely.  However there may be a segment inside that 1 minute period that takes say 45 seconds.  And like you suspected the rank is based on w/kg / time.  So the same power (or even slightly higher) for 45 seconds of that selection will not rank as high as the same power held for 1 minute (or at least evaluated for a minute).  I hope that makes some sense.  

TL;DR;  The rank badges for segments are bound to a fixed time domain.  The time period for the Peaks table and the time taken for each segment in the segments list.

I hope that helps.  Also, if you haven't already, click on the rank badge.  There is a little widget that lets you see more details.

**Comment by mayfield** [comment]
Hi @ph4r Let me know if my reply resolves this for you so I can close this issue or if you wanted to discuss further.

**Comment by ph4r** [comment]
Clear to Close - Your NP weighting makes a big difference and is what pushes my total active time up a Cat.
